test.js

```diff
@@ -4,14 +4,41 @@ import inRange from 'in-range';
 import fn from './';
 
 test('promise', async t => {
+	t.plan(1);
 	const end = timeSpan();
 	await fn(50);
-	t.true(inRange(end(), 30, 70));
+	t.true(inRange(end(), 30, 70), 'should be delayed');
 });
 
 test('thunk', async t => {
+	t.plan(2);
 	const end = timeSpan();
 	const result = await Promise.resolve('foo').then(fn(50));
-	t.true(inRange(end(), 30, 70));
-	t.is(result, 'foo');
+	t.true(inRange(end(), 30, 70), 'should be delayed');
+	t.is(result, 'foo', 'should pass through the value');
+});
+
+test.cb('.reject with two arguments', t => {
+	t.plan(2);
+	const error = new Error('foo');
+	const end = timeSpan();
+	fn.reject(50, error)
```

Comment:
> If you don't mind me asking, why?

Sure :). Because promise chains are just an intermediate step from callback hell to `async`/`await`. It's the future (ES7) and it really makes your code a lot cleaner.

> This seems much more clear.

No, it's not ;).

#### your version
```js
test.cb('.reject with two arguments', t => {
	t.plan(2);
	const error = new Error('foo');
	const end = timeSpan();
	fn.reject(50, error)
		.catch(err => {
			t.true(inRange(end(), 30, 70), 'should be delayed');
			t.is(err, error, 'promise should be rejected with the second argument');
			t.end();
		});
});
```

#### my version
```js
test('delay throws error', async t => {
    const end = timeSpan();
    await t.throws(fn.reject(50, new Error('foo')), 'foo');
    t.true(inRange(end(), 30, 70, 'should be delayed'));
});
```